internal/speakerlist/speakerlist.go

```diff
@@ -0,0 +1,155 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package speakerlist
+
+import (
+	"crypto/sha256"
+	golog "log"
+	"strconv"
+	"time"
+
+	"go.universe.tf/metallb/internal/k8s"
+
+	gokitlog "github.com/go-kit/kit/log"
+	"github.com/hashicorp/memberlist"
+)
+
+// SpeakerList represents a list of healthy speakers.
+type SpeakerList struct {
+	l         gokitlog.Logger
+	client    *k8s.Client
+	stopCh    chan struct{}
+	namespace string
+	labels    string
+
+	// The following fields are nil when memberlist is disabled.
```

Comment:
It is an _optional_ implementation detail. You can use it or not, you can choose in the manifest (see instructions here: https://metallb.universe.tf/release-notes/#version-0-9-2). If memberlist is disabled, those vars are nil.

I guess that was the confusion? Or what do you mean here with "interchangeably"?

Reply:
I understand both "memberlist" and "speakerlist". What I find confusing/inconsistent is that we seem to be referring to this *entire feature* (namely the "fast dead node detection feature") sometimes as "memberlist" and other times as "speakerlist".

For example, let's look at this:

https://github.com/kinvolk/metallb/blob/38bd40820b57c11fb8a38de39b456add61fc3075/internal/speakerlist/speakerlist.go#L28-L43

Here we have a type called `speakerlist.SpeakerList` which uses **memberlist** in its fields. This fact suggests that memberlist is an implementation detail of speakerlist, i.e. we have a feature called "speakerlist" which uses memberlist under the hood.

Now let's look at this:

https://github.com/kinvolk/metallb/blob/38bd40820b57c11fb8a38de39b456add61fc3075/speaker/main.go#L72-L76

The user-facing knobs mention **memberlist** not SpeakerList, which I find inconsistent with the above.

To try and boil this down to a one-liner, I'm trying to understand if we should call the fast dead node detection feature "memberlist" or "speakerlist".

Reply:
Would it make any difference for this PR?

AFAIK, we shouldn't change the user-visible knobs for a patch release (unless is _really_ important). And if the documentation should mention to this feature in one one way or the other, what difference would it make to this PR?

It seems a legit concern, but I don't see why it would make sense to block this PR until then.

Reply:
Okay. Will mark this as solved, thanks!